flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/rules/logical/WindowAggregateReduceFunctionsRule.scala

```diff
@@ -39,14 +40,26 @@ import scala.collection.JavaConversions._
 class WindowAggregateReduceFunctionsRule
   extends AggregateReduceFunctionsRule(
     operand(classOf[LogicalWindowAggregate], any()),
-    RelFactories.LOGICAL_BUILDER) {
+    RelBuilder.proto(
+      Contexts.of(
+        RelFactories.DEFAULT_STRUCT,
+        RelBuilder.Config.DEFAULT
+          .withPruneInputOfAggregate(false)))) {
 
   override def newAggregateRel(
       relBuilder: RelBuilder,
       oldAgg: Aggregate,
       newCalls: util.List[AggregateCall]): Unit = {
 
     // create a LogicalAggregate with simpler aggregation functions
+
+    // Because of:
+    // [CALCITE-3763] RelBuilder.aggregate should prune unused fields from the input,
+    // if the input is a Project.
+    //
+    // the field can not be pruned if it is referenced by other expressions
+    // of the window aggregation(i.e. the TUMBLE_START/END).
+    // To solve this, we config the RelBuilder to forbidden this feature.
```

Comment:
unnecessary comment here?

Reply:
The comment makes the change more clear, so i choose to keep them.